### ISSUE TITLE
Docking on main windows

### ISSUE THREAD

**Issue by maxwellfundi** [IDEMSInternational/R-Instat#3864]
We need to have all the controls with main windows docked because some machines machines with different displays, display the controls differently when they are not docked.

**Comment by dannyparsons** [comment]
I think I can see which ones are not docked correctly from my machine because they display differently in design view. So I can look into this first.

**Comment by rdstern** [comment]
How about the message that the command is taking a long time?  When I use 2 screens with R-Instat on the second, then this is displayed on the other screen.

**Comment by dannyparsons** [comment]
Separate issue, but @maxwellfundi could look into this, I assume it's a start position option.

**Comment by dannyparsons** [comment]
@maxwellfundi have you looked into this?

**Comment by maxwellfundi** [comment]
This is just about the start position and is quite easy to fix. Will have this and the icon done

**Comment by maxwellfundi** [comment]
A solution to @rdstern issue is to make sure the setuploading dialog start position would be centered but then there is a potential problem when this loads for the first time and we need to have the the dialog boxes from R visible. @dannyparsons  is there a way we could bring those dialog boxes from R to the top of everything else?

**Comment by dannyparsons** [comment]
I don't think we have any control over how the RLink displays pop up boxes. We could try changing the Waiting dialog so that it isn't set to Top Most? What are other options?

**Comment by maxwellfundi** [comment]
About this, I think we can have it start in a manually in a position but then that would mean @rdstern  problem is not solved. The other options are center parent which would make it appear on the centre of the data view window as i have noticed. the other is to have it at defaultwindows position which is not great. 

@dannyparsons  my suggestion is get to start in the centerscreen then make a way of making it not the topmost window.

**Comment by maxwellfundi** [comment]
@dannyparsons  something interesting that i think is great is that, we can set the frmsetuploading to center screen but have it topmost to false. All our dialogs have topmost property to false, so the frmsetuploading appears on top of the dialog which would solve rogers issue. Now i am not sure if the dialogs from R when we install R-Instat for the first time will appear topmost since our frmsetuploading  will be false.

**Comment by dannyparsons** [comment]
That sounds like a good idea. But we need to see it with the R windows first as you say, because that is the most important issue.

**Comment by dannyparsons** [comment]
You can test my moving your packages folder to another location and then running R-Instat so that it tries to create it again.

**Comment by dannyparsons** [comment]
That's now resolved, the original issue still has more to do possibly. I have tested the docking on my machine and improved it for any controls and dialogs that needed it. We now want to test an installed version on a machine we know had issues with docking.

I also found a bug in the grid which crashes on close sometimes, although the error isn't visible. Possibly because of now having the grids inside another control. I think this could be resolved by opening each tab on load.
